packages/cli/src/controller/migrate/schema/migrate-schema.controller.ts

```diff
@@ -2,12 +2,221 @@
 // SPDX-License-Identifier: GPL-3.0
 
 import fs from 'fs';
+import {
+  Source,
+  parse,
+  print,
+  TypeNode,
+  NamedTypeNode,
+  Kind,
+  StringValueNode,
+  ValueNode,
+  ObjectValueNode,
+  ObjectTypeDefinitionNode,
+  DefinitionNode,
+  DirectiveNode,
+  ListValueNode,
+} from 'graphql';
 
-export async function migrateSchema(subgraphSchemaPath: string, subqlSchemaPath: string) {
+// Extra schema definitions used by the graph
+// const subgraphScalars = gql`
+//   scalar Int8
+//   scalar Timestamp
+// `;
+
+// const subgraphDirectives = gql`
+//   type FulltextInclude {
+//     name: String!
+//   }
+
+//   directive @fulltext(name: String!, language: String!, algorithm: String!, include: [FulltextInclude!]!) on OBJECT
+// `;
+
+export async function migrateSchema(subgraphSchemaPath: string, subqlSchemaPath: string): Promise<void> {
   await fs.promises.rm(subqlSchemaPath, {force: true});
-  // copy over schema
-  fs.copyFileSync(subgraphSchemaPath, subqlSchemaPath);
+
+  const file = await fs.promises.readFile(subgraphSchemaPath);
+  const output = migrateSchemaFromString(file.toString('utf8'));
+  await fs.promises.writeFile(subqlSchemaPath, output);
   console.log(
-    `* schema.graphql have been copied over, they will need to be updated to work with SubQuery. See our documentation for more details https://academy.subquery.network/build/graphql.html`
+    `* schema.graphql has been migrated. If there are any issues see our documentation for more details https://academy.subquery.network/build/graphql.html`
   );
 }
+
+export function migrateSchemaFromString(input: string): string {
+  const src = new Source(input);
+  const doc = parse(src);
+
+  const updated = doc.definitions
+    .filter((definition) => {
+      if (isObjectType(definition)) {
+        const aggregationDirective = definition.directives?.find((d) => d.name.value === 'aggregation');
+        if (aggregationDirective) {
+          console.warn(`The Aggregation directive is not supported. type="${definition.name.value}" has been removed`);
+          return false;
+        }
+      }
+
+      return true;
+    })
+    .map((definition) => {
+      if (isObjectType(definition)) {
+        // Convert fulltext search directives
+        if (definition.name.value === '_Schema_') {
+          definition.directives?.forEach((directive) => {
+            convertFulltextDirective(directive, doc.definitions);
+          });
+          // No mutations to the global schema
+          return definition;
+        }
+
+        // Map field types to known types
+        (definition as any).fields = definition.fields?.map((field) => {
+          modifyTypeNode(field.type, (type) => {
+            // SubQuery only supports ID type for id
+            if (field.name.value === 'id') {
+              (type.name as any).value = 'ID';
+            }
+            if (type.name.value === 'Int8') {
+              (type.name as any).value = 'Int';
```

Comment:
Im unsure if this is the right conversion, it doesn't seem to be a popular use anyway